pandas/tests/frame/test_to_csv.py

```diff
@@ -1131,3 +1102,83 @@ def test_to_csv_with_dst_transitions(self):
             df.to_pickle(path)
             result = pd.read_pickle(path)
             assert_frame_equal(result, df)
+
+    def test_to_csv_quoting(self):
+        df = DataFrame({
+            'c_string': ['a', 'b,c'],
+            'c_int': [42, np.nan],
+            'c_float': [1.0, 3.2],
+            'c_bool': [True, False],
+        })
+
+        expected = """\
```

Comment:
Style nit, you can use

```
expected = textwrap.dedent(
    """\
    ,c_bool,c_float,c_int,c_string
    0,True,1.0,42.0,a
    1,False,3.2,,"b,c"
    """
)
```

to avoid the odd indentation. Not worth holding up a merge though, if you aren't able to change.